users/views.py

```diff
@@ -1,214 +1,129 @@
-                    sym = "@"
-                    if sym in username_or_email:
-                        user = authenticate(
-                            request, email=username_or_email, password=password
-                        )
-                    else:
-                        user = authenticate(
-                            request, username=username_or_email, password=password
-                        )
-                    if user is not None:
-                        login(request, user)
-                        token, _ = Token.objects.get_or_create(user=user)
-                        return Response(
-                            {
-                                "token": str(token.key),
-                                "message": f"{request.user.username} logged in successfully",
-                            },
-                            status.HTTP_200_OK,
-                        )
-                        # user_login.send(sender=settings.AUTH_USER_MODEL, request=request, user=request.user)
-                    else:
-                        raise AuthenticationFailed("Invalid details")
-            else:
-                return Response(
-                    {"error": serializer.errors}, status.HTTP_400_BAD_REQUEST
-                )
-        except Exception as e:
-            raise AuthenticationFailed("Invalid credentials", str(e))
-    return Response(
-        {"message": "This endpoint handles the authentication of created users."},
-        status.HTTP_200_OK,
-    )
+"""API endpoints using Class Base Views"""
 
+class ListAllAccountAPIView(generics.ListAPIView):
+    queryset = Account.objects.all()
+    serializer_class = ListAccountsSerializer
+    filter_backends = [SearchFilter, OrderingFilter]
+    permission_classes = [IsAdminUser]
 
-@login_required
-@api_view(["GET", "POST"])
-@authentication_classes([TokenAuthentication, SessionAuthentication])
-@permission_classes([IsAuthenticated])
-def user_logout(request):
-    if request.method == "POST":
+class AccountRegistrationAPIView(generics.CreateAPIView):
+    queryset = Account.objects.all()
+    serializer_class = UserSerializer
+    permission_classes = [AllowAny]
+    filter_backends = [SearchFilter, OrderingFilter]
+    
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.create(serializer.validated_data)
+        user_created.send(sender=user.__class__, instance=user)
+        return Response({'message': 'User Created Successfully!', 'account': UserSerializer(user).data}, status.HTTP_201_CREATED)
+        
+     
+    
+class AccountLoginAPIView(APIView):
+    permission_classes = [AllowAny]
+    authentication_classes = [TokenAuthentication, SessionAuthentication]
+    def post(self, request):
+        serializer = LoginSerializer(data=request.data)
+        
+        if serializer.is_valid(raise_exception=True):
```

Comment:
You don't need to use an `if` statement. Just do `serializer.is_valid(raise_exception=True)`

Reply:
Oh right, I’ll check it out